src/main/java/com/ibm/cloud/sdk/core/security/IamAuthenticator.java

```diff
@@ -361,13 +348,14 @@ protected void init(String apikey, String url, String clientId, String clientSec
 
   @Override
   public void validate() {
+    super.validate();
 
-    if (StringUtils.isEmpty(this.url)) {
+    if (StringUtils.isEmpty(this.getURL())) {
       // If no base URL was configured, then use the default IAM base URL.
-      this.url = DEFAULT_IAM_URL;
-    } else if (this.url.endsWith(OPERATION_PATH)) {
+      this.setURL(DEFAULT_IAM_URL);
+    } else {
       // Canonicalize the URL by removing the operation path from it if present.
-      this.url = this.url.substring(0, this.url.length() - OPERATION_PATH.length());
+      this.setURL(StringUtils.removeEnd(this.getURL(), OPERATION_PATH));
```

Comment:
Note: StringUtils.removeEnd() will only remove OPERATION_PATH from the end of the string if the string in fact ends with OPERATION_PATH :)